app/models/scraper.rb

```diff
@@ -42,6 +42,16 @@ def all_watchers
     (watchers + owner.watchers).uniq
   end
 
+  def downloads
+    ApiQuery.where(scraper_id: id)
+  end
+
+  def downloaders
+    downloads.map do |d|
```

Comment:
It will super memory hungry and slow if all the `api_queries` get loaded into memory. Really, you want to use something like
```
downloads.group(:owner_id)
```
which will only return records with different owners. So, you can drop call to `uniq` as well.

Also, to simplify the creation of owner objects add an association on the `ApiQuery` class so that you can just call `owner` on an `ApiQuery` object and get the owner.